### PR TITLE
py-fenics-dolfinx: update for v0.7.2

### PR DESCRIPTION
**Pull request by garth-wells**
Requires #41393.

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the new version sha256, rest LGTM but defer to @adamjstewart

File: var/spack/repos/builtin/packages/py-fenics-dolfinx/package.py [hunk 1]
```diff
@@ -30,34 +31,40 @@ class PyFenicsDolfinx(PythonPackage):
...
 
     depends_on("fenics-dolfinx@main", when="@main")
+    depends_on("fenics-dolfinx@0.7.2", when="@0.7.2")
     depends_on("fenics-dolfinx@0.6.0", when="@0.6.0")
     depends_on("fenics-dolfinx@0.5.1", when="@0.5.1")
     depends_on("fenics-dolfinx@0.5.0", when="@0.5.0")
     depends_on("fenics-dolfinx@0.4.1", when="@0.4.1")
 
     depends_on("fenics-basix@main", type=("build", "link"), when="@main")
+    depends_on("fenics-basix@0.7", type=("build", "link"), when="@0.7")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fenics-dolfinx/package.py:42]: I don't see fenics-basix listed in setup.py

**Review by adamjstewart — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-fenics-dolfinx/package.py (hunk 1 above)
Comment (garth-wells) [reply var/spack/repos/builtin/packages/py-fenics-dolfinx/package.py:42]: `fenics-basix` is a C++ library required for building `py-fenics-dolfinx`, see `CMakeLists.txt` line https://github.com/FEniCS/dolfinx/blob/d685d00d682d0f7addb676e35c891e8a05412c7c/python/CMakeLists.txt#L34.

**Review by chrisrichardson — APPROVED** [review]

**Comment by garth-wells** [comment]
@adamjstewart the PRs that this one depends on have all been merged now. Anything else needed for this PR?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**